federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -141,7 +151,13 @@ func (cc *ClusterController) UpdateClusterStatus() error {
 	for _, cluster := range clusters.Items {
 		if !cc.knownClusterSet.Has(cluster.Name) {
 			glog.V(1).Infof("ClusterController observed a new cluster: %#v", cluster)
+			// add cluster name into knownClusterSet and add cluster client into clusterKubeClientMap
```

Comment:
How about calling addToClusterSet here?
You can also move the glog.V(1) new cluster observed to that function